substrate-node/pallets/pallet-dao/src/lib.rs

```diff
@@ -381,18 +381,13 @@ pub mod pallet {
             let voting = Self::voting(&proposal_hash).ok_or(Error::<T>::ProposalMissing)?;
             ensure!(voting.index == proposal_index, Error::<T>::WrongIndex);
 
-            // Only allow actual closing of the proposal after the voting period has ended.
-            ensure!(
-                frame_system::Pallet::<T>::block_number() >= voting.end,
-                Error::<T>::TooEarly
-            );
-
             let no_votes = voting.nays.len() as u32;
             let yes_votes = voting.ayes.len() as u32;
 
+            // Only allow actual closing of the proposal after the voting threshold is met or voting period has ended
             ensure!(
-                (no_votes + yes_votes) >= voting.threshold,
-                Error::<T>::VoteThresholdNotMet
+                (no_votes + yes_votes) >= voting.threshold || frame_system::Pallet::<T>::block_number() >= voting.end,
```

Comment:
whilst the logic looks good, I think we should rather implement https://github.com/threefoldtech/tfchain/issues/372

Meaning, a proposal can be closed whenever the voting threshold is met. We don't care about the duration anymore